### ISSUE TITLE
Use chroot to clean build (also required for some lib32 packages)

### ISSUE THREAD

**Issue by Xdavius** [pacstall/pacstall#1241]
### Is your feature related to a problem? Please describe.

Some i386 cannot be crossbuild. They need to be in true i386 debootstrap.

### A clear and concise description of what you want to happen.

- schroot Can replace bubblewrap (?)
- use arch-chroot from arch-install-script as alternative to schroot/chroot (need to be tested, I don't know what is the best option)
- Avoid the makedepends to be installed on the host (and to clean them manually), can be cleaned cleaned or not by asking the user to preserve abusive download.
- can be apt update and apt upgrade 
- pacstall install himself into the bootstrap 

Useful for build() function, but all the pacscript can be build inside the chroot for safety.

### Describe alternatives you've considered

As alternative:

pkgname=mypackage
makedepends=(debootstrap schroot binfmt-support)
depends=(real depends for the package)
pacdeps=(add runtime pacdeps)

# create a build.sh containing real makedepends, pacdeps and content of build()
source=("mysource.tar.gz build.sh")

prepare()
debootstrap to ${srcdir} (set --foreign --arch=i386 for lib32)
chroot -c pacstall install (allow execute command inside the chroot from outside)
for make pacdeps 

build()
chroot -c bash build.sh

package()

### Additional context

The alternative solution is an awfull workaround because for each build, you need to recreate a bootstrap and redownload everything. It increases a lot build time and will be a problem for people with slow connection or Gb limited like 4g/5g.

**Comment by oklopfer** [comment]
bwrap is significantly safer, and serves a similar purpose. creating a chroot does not make sense imo. @Elsie19 ?

**Comment by Elsie19** [comment]
Correct.

**Comment by Elsie19** [comment]
Actually coming back to this, I misunderstood the question. A clean build chroot would be interesting because the system we have right now isn't a clean build, it's just sandboxed. I think Xdavius is asking for a clean build sandbox.

**Comment by Xdavius** [comment]
Yes. The ability to set `cleanchroot=true`, or an option for pacstall line `pacstall -I --chroot --i386` for example, will deploy a podman container or whatever to build inside it all arch supported by the package.

Right now, the replaces array is fixed, but there is still a problem with the foreign architecture. Some packages have to be installed twice in 2 architectures on the system (hello libs32... Or better saying foreign i386 architecture).

Right now, if a $pkgname is installed for $amd64, same package name, even from apt reporitory, cannot be installed (And I trigger the pin -1, or the way conflicts works or replaces... In the case of multiarch.

I have found a line in control files for multiarch but I don't really know how it works...

Some packages doesn't have crossbuild packages like llvm and need to be build in the good arch chroot. The resulting package if seted up correctly can be used in a foreign arch environment and also in native.

In part this can be solved with PPR...
